src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -25,152 +26,99 @@ impl HtmlHandlebars {
-                            debug!("[*]: index.html");
-
-                            let mut content = String::new();
-
-                            let _source = File::open(
-                                book.get_destination()
-                                    .expect("If the HTML renderer is called, one would assume the HtmlConfig is set... (3)")
-                                    .join(&ch.path.with_extension("html"))
-                            )?.read_to_string(&mut content);
-
-                            // This could cause a problem when someone displays
-                            // code containing <base href=...>
-                            // on the front page, however this case should be very very rare...
-                            content = content
-                                .lines()
-                                .filter(|line| !line.contains("<base href="))
-                                .collect::<Vec<&str>>()
-                                .join("\n");
-
-                            book.write_file("index.html", content.as_bytes())?;
-
-                            info!("[*] Creating index.html from {:?} ✓",
-                                  book.get_destination()
-                                      .expect("If the HTML renderer is called, one would assume the HtmlConfig is set... (4)")
-                                      .join(&ch.path.with_extension("html"))
-                            );
-                            index = false;
-                        }
+                    // Update the context with data for this file
+                    let path = ch.path
+                        .to_str()
+                        .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "Could not convert path to str"))?;
+
+                    data.insert("path".to_owned(), json!(path));
+                    data.insert("content".to_owned(), json!(content));
+                    data.insert("chapter_title".to_owned(), json!(ch.name));
+                    data.insert("path_to_root".to_owned(), json!(utils::fs::path_to_root(&ch.path)));
+
+                    // Render the handlebars template with the data
+                    debug!("[*]: Render template");
+                    let rendered = handlebars.render("index", &data)?;
+                    let rendered = self.post_process(rendered);
+
+                    let filename = Path::new(&ch.path).with_extension("html");
+
+                    // Write to file
+                    info!("[*] Creating {:?} ✓", filename.display());
+                    book.write_file(filename, &rendered.into_bytes())?;
+
+                    // Create an index.html from the first element in SUMMARY.md
+                    if *index {
+                        debug!("[*]: index.html");
+
+                        let mut content = String::new();
+
+                        let _source = File::open(destination.join(&ch.path.with_extension("html")))
```

Comment:
Oops! That's probably a bug in `rustfmt-nightly` and I didn't take the time to look over the code for formatting issues before committing. 

I think the `_source` was left over from the original `HtmlHandlebars` implementation and I didn't notice it wasn't being used because the leading underscore silences unused variable warnings.